File: 백준/Bronze/2798. 블랙잭/블랙잭.py
N,M = map(int, input().split())
cards = list(map(int,input().split()))

target = M
sub = M
result = 0

for i in range (N-2):
    for j in range (i+1,N-1):
        for k in range (j+1,N):
            num = cards[i] + cards[j] + cards[k]
            if num <= target:
                newsub = target - num
                if newsub < sub:
                    sub = newsub
                    result = num

print(result)

File: 백준/Bronze/2798. 블랙잭/블랙잭.c
#include <stdio.h>
#include <stdlib.h>

int main(void){
    int M, N;
    scanf("%d %d", &N, &M);

    int * arr = (int*)malloc(sizeof(int)*N);

    for(int i=0; i<N; i++){
        scanf("%d", &arr[i]);
    }

    int sub = M;
    int result = 0;

    for(int i=0; i<N-2; i++){
        for(int j=i+1; j<N-1; j++){
            for(int k=j+1; k<N; k++){
                int num = arr[i] + arr[j] + arr[k];
                if (num <= M){
                    int newsub = M - num;
                    if (newsub < sub){
                        sub = newsub;
                        result = num;
                    }
                }
            }
        }
    }

    printf("%d", result);

    return 0;

}
